src/functional/iterator/embedded.py

```diff
@@ -187,6 +187,11 @@ def or_(a, b):
 
 @builtins.tuple_get.register(EMBEDDED)
 def tuple_get(i, tup):
+    if isinstance(tup, tuple):
+        return tup[i]
+    assert isinstance(tup, (np.ndarray, np.void))
+    if tup.dtype.names:
+        return tup[tup.dtype.names[i]]
     return tup[i]
```

Comment:
is this line reachable?

Reply:
Yes, it’s for fields with tuple dimensions. Clarified in the comments.